### DISCUSSION TITLE
tutorials/scripting/c_sharp/c_sharp_features

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#274] [User-contributed notes]
# tutorials/scripting/c_sharp/c_sharp_features

This page provides an overview of the commonly used features of both C# and Godot and how they are used together. Type conversion and casting: C# is a statically typed language. Therefore, you can'...

https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_features.html

**Comment by copyrat90** [comment]
It seems that `GODOT_64` and `GODOT_32` constants are missing?
```csproj
<Target Name="ConstantTestMessage" AfterTargets="Build">
  <Message Text="DefineConstants = $(DefineConstants)"/>
</Target>
```
```powershell
DefineConstants = GODOT;GODOT_WINDOWS;GODOT_PC;GODOT4;GODOT4_4;GODOT4_4_1;GODOT4_OR_GREATER;GODOT4_0_OR_GREATER;GODOT4_1_OR_GREATER;GODOT4_2_OR_GREATER;GODOT4_3_OR_GREATER;GODOT4_4_OR_GREATER;GODOT4_4_0_OR_GREATER;GODOT4_4_1_OR_GREATER;TRACE;TOOLS;DEBUG;NET;NET9_0;NETCOREAPP;NET5_0_OR_GREATER;NET6_0_OR_GREATER;NET7_0_OR_GREATER;NET8_0_OR_GREATER;NET9_0_OR_GREATER;NETCOREAPP1_0_OR_GREATER;NETCOREAPP1_1_OR_GREATER;NETCOREAPP2_0_OR_GREATER;NETCOREAPP2_1_OR_GREATER;NETCOREAPP2_2_OR_GREATER;NETCOREAPP3_0_OR_GREATER;NETCOREAPP3_1_OR_GREATER
```

**Reply by raulsntos** [reply]
Nice catch. A number of these defines from Godot 3 were removed in 4 and we missed them when we updated the documentation. I opened a PR to fix it: https://github.com/godotengine/godot-docs/pull/10846

**Reply by Joy-less** [reply]
I opened an issue relating to this over a year ago, but it was never fixed:
https://github.com/godotengine/godot-docs/issues/9230
